M2/Macaulay2/d/expr.d

```diff
@@ -35,10 +35,12 @@ export nextHash():hash_t := (
      HashCounter);
 
 -- Knuth, Art of Computer Programming, Section 6.4
+-- constant is approx. 2^64 / phi
 export fibonacciHash(k:hash_t,p:int):hash_t := (
-    Ccode(hash_t, "(2654435769 * ",k,") >> (32 - ",p,")"));
+    Ccode(hash_t, "(11400714819323198485ull * ",k,") >> (64 - ",p,")"));
 
 -- hash codes for mutable objects that don't use nextHash
+-- we use 9 since #buckets ZZ = 2^9
```

Comment:
Why is `#buckets ZZ` relevant here? `#buckets Matrix = 2^10`.

Reply:
It was just an arbitrary decently large hash table.  The algorithm says that if $2^n$ is the number of bits in our hash code, multiply $2^n/\varphi$ by the input, and then shift that by $n - m$, where $2^m$ is the size of the hash table.  But our hash tables vary in size and we won't know which hash table we might be stuffing this thing into, so I just picked `ZZ`.